src/signals-unix.c

```diff
@@ -780,6 +780,9 @@ static void *signal_listener(void *arg)
                     }
                     jl_set_safe_restore(old_buf);
 
+                    // store threadid but add 1 as 0 is preserved to indicate end of block
```

Comment:
You can access the thread local state with:

```
    for (int i = 0; i < jl_n_threads; i++) {
        jl_ptls_t ptls2 = jl_all_tls_states[i];
```

And then:

```
ptls->tid 
ptls->current_task
```

Reply:
thanks. Added